app/scripts/modules/amazon/src/serverGroup/AvailabilityZoneSelector.tsx

```diff
@@ -39,8 +39,8 @@ export class AvailabilityZoneSelector extends React.Component<
   private setDefaultZones(props: IAvailabilityZoneSelectorProps) {
     const { credentials, onChange, region } = props;
 
-    AccountService.getAvailabilityZonesForAccountAndRegion('aws', credentials, region).then(preferredZones =>
-      onChange(preferredZones.slice()),
+    AccountService.getAvailabilityZonesForAccountAndRegion('aws', credentials, region).then(
+      preferredZones => preferredZones && onChange(preferredZones.slice()),
```

Comment:
Would wrap the function body here as `{ preferred Zones && onChange(preferredZones.slice()) }`. As is, the value of the promise (even though it's not being used) is either whatever falsey value `preferredZones` is or the return type of `onChange`.

Reply:
Should I add this as a subsequent patch on top of the cherry-picked commits?